cli/tests/pcluster/api/controllers/test_cluster_compute_fleet_controller.py

```diff
@@ -5,41 +5,215 @@
 #  or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
```

Comment:
Probably you will want to add some additional test cases to make sure error conditions are properly handled and translated into the appropriate error codes

Reply:
TBD

Reply:
Added unit tests for trying to update compute fleet status while stack is IN_PROGRESS status